x-pack/plugins/monitoring/public/application/pages/elasticsearch/nodes_page.tsx

```diff
@@ -140,3 +156,35 @@ export const ElasticsearchNodesPage: React.FC<ComponentProps> = ({ clusters }) =
     </ElasticsearchTemplate>
   );
 };
+
+function sortNodeRoles(roles: string[]): string[] | undefined {
+  if (!roles) {
+    return undefined;
+  }
+
+  if (roles.length === 0) {
+    return roles;
+  }
+
+  const roleMap: { [key: string]: string } = {};
+  roles.forEach((role) => {
+    roleMap[role] = role;
+  });
+
+  const sortedRoles = [
+    roleMap.master,
+    roleMap.voting_only,
+    roleMap.data,
+    roleMap.data_content,
+    roleMap.data_hot,
+    roleMap.data_warm,
+    roleMap.data_cold,
+    roleMap.data_frozen,
+    roleMap.ingest,
+    roleMap.transform,
+    roleMap.ml,
+    roleMap.remote_cluster_client,
+  ];
+
+  return sortedRoles.filter((role) => role);
```

Comment:
**suggestion:** you could also optimize this function to define (and create its type) the `sortedRoles` list once in the module scope and use it in combination with a `Set`:
```ts
type Role = 'master' | 'voting_only' | ...

const sortedRoles: Role[] = [
  'master',
  'voting_only',
  'data',
  'data_content',
  'data_hot',
  'data_warm',
  'data_cold',
  'data_frozen',
  'ingest',
  'transform',
  'ml',
  'remote_cluster_client',
];

function sortNodeRoles (roles: Role[]): Role[] {
  // ...

  const rolesSet = new Set(roles);

  return sortedRoles.filter(role => rolesSet.has(role))
}
```

Reply:
Great tip! Much cleaner, though I skipped creating the Set since for such a small number of items the performance of `Array.includes` should be enough!